src/ng/rootScope.js

```diff
@@ -1167,7 +1167,7 @@ function $RootScopeProvider() {
       $on: function(name, listener) {
         var namedListeners = this.$$listeners[name];
         if (!namedListeners) {
-          this.$$listeners[name] = namedListeners = [];
+          this.$$listeners[name] = namedListeners = extend([], {$$index: -1});
```

Comment:
I think I'll drop this line and use `undefined` instead of `-1` as the "not looping right now" value

Reply:
+1